src/main/java/com/liveramp/captain/daemon/CaptainJoblet.java

```diff
@@ -188,7 +187,13 @@ private void submitRequest(CaptainRequestConfig config) throws DaemonException {
       notifier.notify(subject, e, CaptainNotifier.NotificationLevel.ERROR);
       requestUpdater.cancel(id);
       return;
-    } catch (Throwable e) {
+    } catch(CaptainSubmissionException ce)
+    {
+      String subject = String.format("%s: handle transient submission failed for request %s", CaptainAlertHelpers.getHostName(), id);
+      notifier.notify(subject, ce, CaptainNotifier.NotificationLevel.ERROR);
+      requestUpdater.pending(id);
```

Comment:
So I actually just noticed that we pick up requests in READY and submit them, not PENDING: https://github.com/LiveRamp/captain/pull/21/files#diff-26f401cbdb54d340dd8e7b70e37b53d5R89

So we will want to keep the status as READY, not update it to PENDING

Reply:
ok

Reply:
Reconfirming your finding from captain readme :)

**READY: When Captain encounters a request in this status, it is going try to run the RequestSubmitter for the step that it's currently in**.

Reply:
Updated it.